v2/internal/reconcilers/arm/azure_generic_arm_reconciler_instance.go

```diff
@@ -65,13 +66,18 @@ func (r *azureDeploymentReconcilerInstance) CreateOrUpdate(ctx context.Context)
 	r.Log.V(Verbose).Info("Reconciling resource", "action", action)
 
 	result, err := actionFunc(ctx)
+	//TODO: check result here and record requeue and move RecordAzureRequestsTotalPUT inside where making put request
+
 	if err != nil {
 		r.Log.Error(err, "Error during CreateOrUpdate", "action", action)
 		r.Recorder.Event(r.Obj, v1.EventTypeWarning, "CreateOrUpdateActionError", err.Error())
-
 		return ctrl.Result{}, err
 	}
 
+	if result.Requeue || result.RequeueAfter > 0 {
```

Comment:
Why not record Total PUTs and delete's in the generic client too? I think with this here, we will overcount PUTs because the fact there's a requeue doesn't mean there was a PUT